spark/src/main/scala/geotrellis/spark/ingest/Tiler.scala

```diff
@@ -1,78 +1,66 @@
 package geotrellis.spark.ingest
 
 import geotrellis.spark._
+import geotrellis.spark.tiling._
 import geotrellis.raster._
 import geotrellis.raster.mosaic._
 import geotrellis.vector._
 import org.apache.spark.rdd._
+import org.apache.spark.SparkContext._
 import scala.reflect.ClassTag
 
 object Tiler {
-  /**
-   * This is a Tiler function constructor that will mosaic and tile an RDD of tiles that have an Extent.
-   * Partial application of this function, given getExtent and createKey, produces a [[Tiler]].
-   *
-   * @param getExtent function to extract extent from source key T
-   * @param createKey function to create target key, K, given corresponding source key, T, and a [[SpatialComponent]]
-   * @param rdd       rdd containing source tiles, may be overlapping
-   * @param metaData  RasterMetaData result tiles
-   * @tparam T        type of input tile key
-   * @tparam K        type of output tile key, must have SpatialComponent
-   * @return          RasterRDD[K] of mosaicked tiles
-   */
-  def apply[T, K: SpatialComponent: ClassTag]
-    (getExtent: T=> Extent, createKey: (T, SpatialKey) => K)
-    (rdd: RDD[(T, Tile)], metaData: RasterMetaData)
-  : RasterRDD[K] = {
-
-    val bcMetaData = rdd.sparkContext.broadcast(metaData)
-
-    val tilesWithKeys: RDD[(K, (K, Extent, Tile))] =
-      rdd
-        .flatMap { { tup =>
+  def cutTiles[T, K: SpatialComponent: ClassTag] (
+    getExtent: T=> Extent,
+    createKey: (T, SpatialKey) => K,
+    rdd: RDD[(T, Tile)],
+    mapTransform: MapKeyTransform,
+    cellType: CellType,
+    tileLayout: TileLayout
+  ): RDD[(K, Tile)] =
+    rdd
+      .flatMap { tup =>
         val (inKey, tile) = tup
-        val metaData = bcMetaData.value
         val extent = getExtent(inKey)
-        metaData.mapTransform(extent)
+        mapTransform(extent)
           .coords
           .map  { spatialComponent =>
-          val outKey = createKey(inKey, spatialComponent)
-          (outKey, (outKey, extent, tile))
-        } }
-      }
-
-    // Functions for combine step
-    val createTile =  { (tup: (K, Extent, Tile)) => {
-      val (key, extent, tile) = tup
-      //val metaData = bcMetaData.value
-      val tmsTile = ArrayTile.empty(metaData.cellType, metaData.tileLayout.tileCols, metaData.tileLayout.tileRows)
-      tmsTile.merge(metaData.mapTransform(key), extent, tile)
-    }}
+            val outKey = createKey(inKey, spatialComponent)
+            val tmsTile = ArrayTile.empty(cellType, tileLayout.tileCols, tileLayout.tileRows)
+            tmsTile.merge(mapTransform(outKey), extent, tile)
```

Comment:
This is a lot cleaner looking but does it sacrifice performance? Now it's blowing up the flatMap into a set of actual tiles rather than references to original tile.